app/src/components/UI/Molecules/LandingPageSidebar/LandingPageSidebar.tsx

```diff
@@ -52,30 +89,83 @@ const LandingPageSidebar: React.FC = () => {
         />
         <h3 className={styles.header}>Filters</h3>
         <h3 className={styles.header}>Cuisine</h3>
-        <SelectFilter />
+
+        <SelectFilter
+          addSelection={handleChange}
+          selections={cuisineSelection}
+        />
+
         <h3 className={styles.header}>Prices</h3>
-        <Checkbox label="0~5$" className={styles.checkbox} />
-        <Checkbox label="5~10$" className={styles.checkbox} />
-        <Checkbox label="10~15$" className={styles.checkbox} />
-        <Checkbox label="20+$" className={styles.checkbox} />
+        <Checkbox
+          label="Cheap"
+          className={styles.checkbox}
+          onChange={(e, data) => handleCheapSelectionChange(e, data.checked)}
+        />
+
+        <Checkbox
+          label="Medium"
+          className={styles.checkbox}
+          onChange={(e, data) => handleMediumSelectionChange(e, data.checked)}
+        />
+
+        <Checkbox
+          label="Gourmet"
+          className={styles.checkbox}
+          onChange={(e, data) => handleGourmetSelectionChange(e, data.checked)}
+        />
       </Menu.Item>
       <Menu.Item>
         <h3 className={styles.header}>Results</h3>
       </Menu.Item>
+
       <Container textAlign="left">
         {resultVendors
-          ? resultVendors.map((vendor) => (
-              <Container key={vendor.ID}>
-                <Container className={styles.vendorInfo}>
-                  <h2>
-                    <Link to={`/vendors/${vendor.ID}`}>{vendor.Name}</Link>
-                  </h2>
-                  <p>Address: {vendor.BusinessAddress}</p>
-                  <p>Business Hours: {vendor.BusinessHours}</p>
-                </Container>
-                <Container className={styles.divider} />
-              </Container>
-            ))
+          ? resultVendors.map((vendor) => {
+              if (
```

Comment:
Although this works, we should not be filtering results on the frontend side. We wrote an OpenSearch query to filter results. The main issue with your approach is that if we had thousands of search results, this would make the search response very slow. The response from OpenSearch will only contain results that are filtered.